src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -531,16 +577,18 @@ impl Net {
     }
 
     // We currently prioritize packets from the MMDS over regular network packets.
-    fn read_from_mmds_or_tap(&mut self) -> Result<Option<ParsedDescriptorChain>, NetError> {
-        // If we don't have any buffers available try to parse more from the RX queue. There might
-        // be some buffers we didn't get the chance to process, because we got to handle the TAP
-        // event before the RX queue event.
-        if self.rx_buffer.is_empty() {
+    fn read_from_mmds_or_tap(&mut self) -> Result<Option<NonZeroU32>, NetError> {
+        // We only want to read from TAP (or mmds) if we have at least 64K of available capacity as
+        // this is the max size of 1 packet.
+        // SAFETY:
+        // * MAX_BUFFER_SIZE is constant and fits into u32
+        #[allow(clippy::cast_possible_truncation)]
```

Comment:
I really dislike `try_into().unwrap()` (after Patrick brainwashing me :P) but I dislike this more :) I know that there are places that the truncation cannot happen today (like this one) but 1. you use them all over the place and 2. I'm afraid of what might happen when/if the code changes in a way that makes it possible to actually truncate. This is like leaving an Easter egg for someone in the future.

Reply:
I think for some of these, we can sidestep the issue by upcasting the u32 to usize for the comparison, instead of coercing the constant (oh how I wish clippy was clever enough to understand that constants will not truncate)

Reply:
swapped to use `try_into().unwrap()`.
In general it is not a clippy issue, but Rust issue that it cannot understand compile time integers.

Reply:
Well, even if Rust could at compile time emit an error if the cast of the constant would cause a truncation (which would be neat), there's still the problem that clippy's `cast_possible_truncation` lint is "stupid" in the sense that it doesnt try to do any const evaluation. So even if rustc determines "yes this is fine", clippy would also need to be updated to specifically ignore const-casts that rustc determined are fine.